sdcm/utils/sstable/sstable_utils.py

```diff
@@ -131,39 +131,30 @@ def check_that_sstables_are_encrypted(self, sstables=None,  # pylint: disable=to
             f" Success part: '{encryption_success_part}'. Expected bool value: '{expected_bool_value}'."
             f" Encryption results: {encryption_results}")
 
-    def count_sstable_tombstones(self, sstable: str) -> int:
+    def _count_ttl_expired_tombstones(self, sstable: str) -> int:
         """
-        Counts the number of tombstones in a given SSTable.
+        Counts the number of ttl-expired tombstones in a given SSTable.
 
         :param sstable: The SSTable file path.
-        :return: The number of tombstones in the SSTable, or 0 if SSTable doesn't exist.
+        :return: The number of ttl-expired tombstones in the SSTable, or 0 if SSTable doesn't exist.
         """
         if not self._run_sstabledump(sstable=sstable):  # Check if SSTable exists and was dumped
             self.log.debug("Skipping tombstone count as SSTable %s does not exist or dump failed.", sstable)
             return 0
 
-        # Fetch the dumped JSON content
         result = self.db_node.remoter.run(
-            f'sudo cat {self.REMOTE_SSTABLEDUMP_PATH}', verbose=False, ignore_status=False)
-
-        if not result.ok:
-            self.log.debug("Failed to retrieve SSTable dump data for %s: (%s, %s)", sstable, result.stdout,
-                           result.stderr)
-            return 0
-
-        try:
-            dump_data = json.loads(result.stdout)
-            num_tombstones = sum(
-                1 for partition in dump_data.get("sstables", {}).get("anonymous", [])
-                if "tombstone" in partition or partition.get("expired") is True
-            )
-
-            self.log.debug("Found %s tombstones in SSTable %s", num_tombstones, sstable)
+            f"sudo awk -F 'deletion_time' '{{print NF-1}}' {self.REMOTE_SSTABLEDUMP_PATH}",
```

Comment:
Why use `awk` (and why with `sudo`), when we have a structured output (`JSON`) that you can parse with `jq`.
Also, this indiscriminate matching of `deletion_time` might include things which have no TTL, like regular tombstones.
To match only TTL expired tombstones, you need to match only the following entities:
* row marker
* cells -- both regular and collection cells

And nothing else. Higher-order tombstones also have `deletion_time` but they are not subject to TTL and thus would constitute false-positive matches here.

Reply:
ok, so for avoiding confusions and false-positive, the method is renamed "count_tombstones". So currently no differentiation of tombstone type, which is sufficient for this test.
we can consider using "jq" for next steps, but on the other hand it would also be counting more on implementation details.
In any case we better use it first by adding ttl tests to dtest.